UnityProject/Assets/Scripts/Player/IPlayerPossessable.cs

```diff
@@ -10,23 +10,59 @@
 
 public interface IPlayerPossessable
 {
-
 	public GameObject GameObject { get; }
-	public IPlayerPossessable Possessing { get; set; }
 
-	public GameObject PossessingObject { get; set; }
+	private GameObject GetPossessingObject()
+	{
+		if (PossessingID is NetId.Empty or NetId.Invalid)
+		{
+			return null;
+		}
+		else
+		{
+			var spawned = CustomNetworkManager.IsServer ? NetworkServer.spawned : NetworkClient.spawned;
+			return spawned[PossessingID].gameObject;
+		}
+	}
+
+	public IPlayerPossessable GetPossessing()
+	{
+		var ob = GetPossessingObject();
+		if (ob != null)
+		{
+			return ob.GetComponent<IPlayerPossessable>();
+		}
+
+		return null;
+	}
+
+	public uint PossessingID { get; }
 
 	public Mind PossessingMind { get; set; }
 
 	public IPlayerPossessable PossessedBy { get; set; }
 
-	public MindNIPossessingEvent OnPossessedBy  { get; set; }
+	public MindNIPossessingEvent OnPossessedBy { get; set; }
```

Comment:
don't put variables in the middle or below functions. put them all at the top of the script.